src/Symfony/Component/HttpClient/HttpClient.php

```diff
@@ -61,6 +61,8 @@ public static function create(array $defaultOptions = [], int $maxHostConnection
             return new AmpHttpClient($defaultOptions, null, $maxHostConnections, $maxPendingPushes);
         }
 
+        @trigger_error(($curl ? 'Upgrade' : 'Install').' the curl extension or run "composer require amphp/http-client" to perform async HTTP operations, including full HTTP/2 support', E_USER_NOTICE);
```

Comment:
instead of using a variable, I would put `\extension_loaded('curl')` inline again. That function call is resolved at compile-time by OPCache, but the variable value is not.